Solutions/Marain.UserNotifications.Abstractions/Marain/NotificationTemplate/NotificationTemplate.cs

```diff
@@ -0,0 +1,65 @@
+﻿// <copyright file="NotificationTemplate.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.NotificationTemplate.NotificationTemplate
+{
+    using System;
+    using Marain.NotificationTemplate.NotificationTemplate.CommunicationTemplates;
+
+    /// <summary>
+    /// The Notification Template structure.
+    /// </summary>
+    public class NotificationTemplate
+    {
+        /// <summary>
+        /// Constructor for the notification template object.
+        /// </summary>
+        /// <param name="notificationType">The notification type.</param>
+        /// <param name="timestamp">The last updated time.</param>
+        /// <param name="smsTemplate">The SMS object for the template.</param>
+        /// <param name="emailTemplate">The email object for the notification.</param>
+        /// <param name="webPushTemplate">The web push object for the notification.</param>
+        public NotificationTemplate(
+            string notificationType,
+            DateTimeOffset? timestamp = null,
```

Comment:
I'm always a bit wary of these timestamps? I can see them in a few places. What are they being used for? And are they safe against clock drift?

I don't see anything that looks like ETag support in here, which is likely to be the kind of thing we need for concurrency management, rather than Timestamps - unless there is some other purpose behind them.

Reply:
We have changed the implementation for user preferences and notification templates both to use eTags.